### ISSUE TITLE
Arrays in additionalProperties are ignored

### ISSUE THREAD

**Issue by Echronix** [koxudaxi/datamodel-code-generator#229]
**Describe the bug**
When using an object with additionalProperties and an array of an item type, the resulting code doesn't reference the generated item type.

**To Reproduce**

Example schema:
```json

openapi: "3.0.0"
info:
  title: Sample API
  description: API description in Markdown.
  version: 1.0.0
servers: []
paths: {}
components:
  schemas:
    test:
      type: object
      properties:
        broken:
          type: object
          additionalProperties:
            type: array
            items:
              $ref: '#/components/schemas/broken'
    broken:
      type: object
      properties:
        foo:
          type: string
        bar:
          type: integer

```
Results in the following generated file:
```py
# generated by datamodel-codegen:
#   filename:  tmp.yml
#   timestamp: 2020-09-30T13:47:06+00:00

from __future__ import annotations

from typing import Dict, Optional

from pydantic import BaseModel


class Broken(BaseModel):
    foo: Optional[str] = None
    bar: Optional[int] = None


class Test(BaseModel):
    broken: Optional[Dict[str, Broken]] = None
```

Used commandline:
```
$ datamodel-codegen --input test.yml --input-file-type openapi --output test_model.py
```

**Expected behavior**

```py
class Test(BaseModel):
    broken: Optional[Dict[str, List[Broken]]] = None
```

**Version:**
- OS: Linux
- Python Verison: 3.8.5
- datamodel-code-generator Version: 0.5.30

**Comment by koxudaxi** [comment]
@Echronix 
Thank you for creating this issue.
OK, I will implement it.

**Comment by languitar** [comment]
I just hit this as well again

**Comment by koxudaxi** [comment]
I have released a new version as  `0.5.39` :rocket:
